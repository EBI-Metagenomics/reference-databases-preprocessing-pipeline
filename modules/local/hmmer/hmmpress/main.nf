process HMMER_HMMPRESS {
    tag "$meta.id"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://community-cr-prod.seqera.io/docker/registry/v2/blobs/sha256/07/07c4cbd91c4459dc86b13b5cd799cacba96b27d66c276485550d299c7a4c6f8a/data' :
        'community.wave.seqera.io/library/hmmer:3.4--cb5d2dd2e85974ca' }"

    input:
    tuple val(meta), path(hmmfile)

    output:
    tuple val(meta), path("*.h3?"), emit: compressed_db
    path "versions.yml"           , emit: versions

    script:
    def args = task.ext.args ?: ''

    """
    hmmpress \\
        $args \\
        ${hmmfile}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hmmer: \$(echo \$(hmmpress -h | grep HMMER | sed 's/# HMMER //' | sed 's/ .*//' 2>&1))
    END_VERSIONS
    """

    stub:
    def prefix    = task.ext.prefix ?: "stub"

    """
    touch ${prefix}.h3m
    touch ${prefix}.h3i
    touch ${prefix}.h3f
    touch ${prefix}.h3p

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        hmmer: \$(echo \$(hmmbuild -h | grep HMMER | sed 's/# HMMER //' | sed 's/ .*//' 2>&1))
    END_VERSIONS
    """
}
