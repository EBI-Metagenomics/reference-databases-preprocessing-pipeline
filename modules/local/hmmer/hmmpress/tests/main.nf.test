
nextflow_process {

    name "Test Process HMMER_HMMPRESS"
    script "../main.nf"
    process "HMMER_HMMPRESS"

    tag "modules"
    tag "modules_nfcore"
    tag "hmmer"
    tag "hmmer/hmmpress"

    test("test-hmmer-hmmpress") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file("${moduleDir}/tests/data/profiles.hmm", checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("test-hmmer-hmmpress-stub") {
        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test' ],
                    file('https://raw.githubusercontent.com/nf-core/test-datasets/modules/data/delete_me/hmmer/bac.16S_rRNA.hmm.gz', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

}
