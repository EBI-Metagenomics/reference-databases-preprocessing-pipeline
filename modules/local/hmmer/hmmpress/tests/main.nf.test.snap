{
    "test-hmmer-hmmpress": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "profiles.hmm.h3f:md5,972a29c1c1c783f2016b83437f76adad",
                            "profiles.hmm.h3i:md5,e11925c741a1799d54e2f2be58726797",
                            "profiles.hmm.h3m:md5,3196ccc515883d4c4ce264fc75c7d148",
                            "profiles.hmm.h3p:md5,6e1e1c56020046cd43de0a6520fc2170"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,f5473ee4ad53142d92c79b5a0fe94bf6"
                ],
                "compressed_db": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "profiles.hmm.h3f:md5,972a29c1c1c783f2016b83437f76adad",
                            "profiles.hmm.h3i:md5,e11925c741a1799d54e2f2be58726797",
                            "profiles.hmm.h3m:md5,3196ccc515883d4c4ce264fc75c7d148",
                            "profiles.hmm.h3p:md5,6e1e1c56020046cd43de0a6520fc2170"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f5473ee4ad53142d92c79b5a0fe94bf6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-21T17:00:20.421917"
    },
    "test-hmmer-hmmpress-stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "stub.h3f:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "stub.h3i:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "stub.h3m:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "stub.h3p:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,f5473ee4ad53142d92c79b5a0fe94bf6"
                ],
                "compressed_db": [
                    [
                        {
                            "id": "test"
                        },
                        [
                            "stub.h3f:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "stub.h3i:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "stub.h3m:md5,d41d8cd98f00b204e9800998ecf8427e",
                            "stub.h3p:md5,d41d8cd98f00b204e9800998ecf8427e"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,f5473ee4ad53142d92c79b5a0fe94bf6"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.0"
        },
        "timestamp": "2025-02-21T17:00:25.389552"
    }
}